fiftyone/core/dataset.py

```
@@ -1210,26 +1210,43 @@ def _frame_collstats(self):
 
         Returns:
             a :class:`fiftyone.core.sample.Sample`
         """
+        cursor = self._get_first(reverse=True)
+
         try:
-            sample_view = self[-1:].first()
-        except ValueError:
+            d = next(cursor)
+        except StopIteration:
             raise ValueError("%s is empty" % self.__class__.__name__)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)